chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/AuthenticationController.java

```diff
@@ -59,21 +76,142 @@ public class AuthenticationController {
+    public AuthenticationController(AuthenticationManager authenticationManager, UserAccountUpdateEmailer userAccountUpdateEmailer,
+            BCryptPasswordEncoder bCryptPasswordEncoder, UserManager userManager, JWTUserConverterFacade userConverterFacade,
+            ErrorMessageUtil msgUtil, CognitoAuthenticationManager cognitoAuthenticationManager, CognitoPasswordManager cognitoPasswordManager,
+            FF4j ff4j) {
         this.authenticationManager = authenticationManager;
         this.userAccountUpdateEmailer = userAccountUpdateEmailer;
         this.bCryptPasswordEncoder = bCryptPasswordEncoder;
         this.userManager = userManager;
         this.userConverterFacade = userConverterFacade;
         this.msgUtil = msgUtil;
+
+        this.cognitoAuthenticationManager = cognitoAuthenticationManager;
+        this.cognitoPasswordManager = cognitoPasswordManager;
+        this.ff4j = ff4j;
     }
 
+
+    @Operation(summary = "Log in.",
+            description = "Call this method to authenticate a user. The value returned is that user's "
+                    + "token which must be passed on all subsequent requests in the Authorization header. "
+                    + "Specifically, the Authorization header must have a value of 'Bearer token-that-gets-returned'.",
+            security = {
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.API_KEY)
+            }
+        )
+    @ApiResponse(responseCode = "470", description = "The user is required to respond to the described challenge.")
+    @RequestMapping(value = "", method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE, produces = "application/json; charset=utf-8")
+    public CognitoAuthenticationResponse authenticate(@RequestBody LoginCredentials credentials) throws CognitoAuthenticationChallengeException, CognitoPasswordResetRequiredException {
+
+        if (!ff4j.check(FeatureList.SSO)) {
+            throw new NotImplementedException("This method has not been implemented");
+        }
+
+        CognitoAuthenticationResponse response = cognitoAuthenticationManager.authenticate(credentials);
+        if (response == null) {
+            throw new ChplAccountStatusException(msgUtil.getMessage("auth.loginNotAllowed"));
+        }
+        return response;
+    }
+
+    @Operation(summary = "Log user out.",
+            description = "Invalidates all of the tokens associated with the user.",
+            security = {
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.API_KEY)
+            }
+        )
+    @RequestMapping(value = "/logout", method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE, produces = "application/json; charset=utf-8")
+    public void logout(@RequestBody CognitoLogoutRequest request) {
+        cognitoAuthenticationManager.invalidateTokensForUser(request.getEmail());
```

Comment:
Does this need a FF4J check? It looks like most of the other endpoints have one. For the ones that don't, should they as well?